src/java/org/apache/cassandra/net/Message.java

```diff
@@ -473,6 +482,13 @@ public Builder<T> withParam(ParamType type, Object value)
             return this;
         }
 
+        public Builder<T> withCustomParam(String name, byte[] value)
+        {
+            params.putIfAbsent(ParamType.CUSTOM_MAP, new HashMap<String,byte[]>());
+            ((Map<String,byte[]>) params.get(ParamType.CUSTOM_MAP)).put(name, value);
```

Comment:
On that note, you are wasting work here by doing the lookup twice - once for `putIfAbsent()`, then again for `get()`.

Not only that, but you are also allocating an entire `HashMap` every time, even if it already exists in `params` map.

Could either use `computeIfAbsent()` and hold on to the return value, or write this as a manual `get()` + `put()` if null.